react-app/src/library/services/vendor-implementations/plantronics/plantronics.ts

```diff
@@ -16,19 +18,41 @@ export default class PlantronicsService extends VendorImplementation {
   deviceIdRetryInterval = 2000;
   vendorName = 'Plantronics';
   pluginName = 'genesys-cloud-headset-library';
+  apiHost = 'https://127.0.0.1:32018/Spokes';
   isActive = false;
   disableEventPolling = false;
   config: ImplementationConfig;
   deviceStatusTimer = null;
   isRetry = false;
   _deviceInfo: DeviceInfo;
+  lastStatusUpdate: Observable<string>;
+  callEventsTimerId: any;
+  deviceStatusTimerId: any;
+
+  private $lastStatusUpdate: BehaviorSubject<string>;
 
   private constructor(config: ImplementationConfig) {
     super(config);
     this.config = config;
-    this.pollForCallEvents();
-    this.pollForDeviceStatus();
     this._deviceInfo = null;
+    this.callEventsTimerId = null;
+    this.deviceStatusTimerId = null;
+    this.$lastStatusUpdate = new BehaviorSubject<string>('');
+    this.lastStatusUpdate = this.$lastStatusUpdate.asObservable();
+    this.lastStatusUpdate.pipe(distinctUntilChanged()).subscribe(value => {
+      switch(value) {
+        case 'isActive':
+          this.pollForCallEvents();
+          break;
+        case 'isConnecting':
+          this.pollForDeviceStatus();
+          break;
+        case 'isConnected':
+        case 'disableEventPolling':
+          this.pollForCallEvents();
+          this.pollForDeviceStatus();
+      }
+    })
```

Comment:
This `switch` statements seems a little unnecessary since both `pollForCallEvents()` and `pollForDeviceStatus()` check the states they care about. Also, this feels odd to be using a behavior subject to trigger theses, but it doesn't look like we are actually using the value emitted from the subject. I think this could probably be re-worked in 1 of 2 ways: 
1. Only use the BehaviorSubject for the state. Meaning, the BehaviorSubject could be `isConnecting | isConnected | isActive | noEvents` (or something similar). Then every time it changed (or even just fired), we would "recheck" whatever state we are currently in. 
2. Keep the service state of `this.isConnecting && this.isConnected && ...` and just use this as a "trigger" to say to check again.